lib/content-services/content-node-selector/content-node-dialog.service.spec.ts

```diff
@@ -103,28 +104,100 @@ describe('ContentNodeDialogService', () => {
         spyOn(sitesService, 'getSites').and.returnValue(Observable.of(fakeSiteList));
         spyOn(documentListService, 'getFolderNode').and.returnValue(Promise.resolve(fakeNode));
         service.openFileBrowseDialogBySite().subscribe();
         tick();
-        expect(materialDialog.open).toHaveBeenCalled();
+        expect(spyOnDialogOpen).toHaveBeenCalled();
     }));
 
     it('should be able to open the dialog for folder using the first user site', fakeAsync(() => {
         spyOn(sitesService, 'getSites').and.returnValue(Observable.of(fakeSiteList));
         spyOn(documentListService, 'getFolderNode').and.returnValue(Promise.resolve(fakeNode));
         service.openFolderBrowseDialogBySite().subscribe();
         tick();
-        expect(materialDialog.open).toHaveBeenCalled();
+        expect(spyOnDialogOpen).toHaveBeenCalled();
     }));
 
     it('should be able to close the material dialog', () => {
         service.close();
         expect(materialDialog.closeAll).toHaveBeenCalled();
     });
 
+    describe('hasEntityCreatePermission', () => {
+        const siteNode: MinimalNodeEntryEntity = <MinimalNodeEntryEntity> {
+            id: 'site',
+            name: 'site-name',
+            nodeType: 'st:site'
+        };
+        const sites: MinimalNodeEntryEntity = <MinimalNodeEntryEntity> {
+            id: 'sites',
+            name: 'sites-name',
+            nodeType: 'st:sites'
+        };
+        const site: MinimalNodeEntryEntity = <MinimalNodeEntryEntity> {
+            id: 'site-id',
+            name: 'site-entry-name',
+            guid: 'any-guid'
+        };
+        const nodeEntryWithRightPermissions: MinimalNodeEntryEntity = <MinimalNodeEntryEntity> {
+            id: 'node-id',
+            name: 'node-name',
+            allowableOperations: ['create']
+        };
+        const nodeEntryNoPermissions: MinimalNodeEntryEntity = <MinimalNodeEntryEntity> {
+            id: 'node-id',
+            name: 'node-name',
+            allowableOperations: []
+        };
+
+        const fixture = [
+            {
+                node: siteNode,
+                infoKey: 'nodeType',
+                expected: false
+            },
+            {
+                node: sites,
+                infoKey: 'nodeType',
+                expected: false
+            },
+            {
+                node: site,
+                infoKey: 'guid',
+                expected: false
+            },
+            {
+                node: nodeEntryWithRightPermissions,
+                infoKey: 'allowableOperations',
+                expected: true
+            },
+            {
+                node: nodeEntryNoPermissions,
+                infoKey: 'allowableOperations',
+                expected: false
+            }
+        ];
+
+        fixture.forEach((testData) => {
+            it(`should be \'${testData.expected}\' for given node with ${testData.infoKey} = \'${testData.node[testData.infoKey]}\'`, () => {
```

Comment:
we don't test method the test has to describe a behaviour, for example should selection be disabled for sites